framework/h5py_interface_creator.py

```diff
@@ -714,267 +426,100 @@ def retrieveAllHistoryNames(self,rootName=None):
       @ In,  rootName, string, optional, It's the root name, if present, only the history names that have this root are going to be returned
       @ Out, workingList, list, List of the HistorySet names
     """
+    if rootName:
+      rname = rootName
     if not self.fileOpen:
       self.__createObjFromFile() # Create the "self.allGroupPaths" list from the existing database
-    workingList = []
-    for index in xrange(len(self.allGroupPaths)):
-      if self.allGroupEnds[self.allGroupPaths[index]]:
-        if rootName and not (rootName in self.allGroupPaths[index].split('/')[1]):
-          continue
-        workingList.append(self.allGroupPaths[index].split('/')[len(self.allGroupPaths[index].split('/'))-1])
+    if not rootName:
+      workingList = [k.split('/')[-1] for k, v in self.allGroupEnds.items() if v ]
+    else:
+      workingList = [k.split('/')[-1] for k, v in self.allGroupEnds.items() if v and k.endswith(rname)]
+
     return workingList
 
-  def retrieveHistory(self,name,filterHist=None,attributes = None):
+  def __getListOfParentGroups(self, grp, backGroups = []):
+    """
+      Method to get the list of groups from the deepest to the root, given a certain group
+      @ In, grp, h5py.Group, istance of the starting group
+      @ InOut, backGroups, list, list of group instances (from the deepest to the root)
+    """
+    if grp.parent and grp.parent != grp:
+      backGroups.append(grp.parent)
+      self.__getListOfParentGroups(grp.parent, backGroups)
+    return backGroups
+
+  def __getNewDataFromGroup(self, group, name):
+    """
+      Get the data from the group
+      @ In, group, h5py.Group, the group from which the data needs to be got
+      @ In, name, str, the group name
+      @ Out, newData, dict, the dictionary with the data
+    """
```

Comment:
Underscores are used in the variable names in this method.

Reply:
fixed